src/modules/displayfood.js

```diff
@@ -0,0 +1,192 @@
+
+/* eslint-disable */
+export const display = async () => {
+/* eslint-enable */
+
+  foodApiObj.forEach(async (item) => {
+    try {
+      const response = await fetch(item);
+      const data = await response.json();
+      if (!response.ok) {
+        main.innerHTML = 'Server Down';
+      }
+
+      data.meals.forEach((item) => {
+        const section = document.createElement('section');
+        const div = document.createElement('div');
+        const btn = document.createElement('button');
+        const btn2 = document.createElement('button');
+        const twoDiv = document.createElement('div');
+        const threeDiv = document.createElement('div');
+        const likeIcon = document.createElement('i');
+        const likeCounts = document.createElement('p');
+
+        section.innerHTML = `
+              <div class="">
+                <span class="" id=${item.idMeal}>
+                </span>
+              </div>
+              <div class="meal_image">
+                <img class="meal_image" src="${item.strMealThumb}" alt="asdf">
+              </div>
+            `;
+        twoDiv.innerHTML = `
+        <p class="meal">${item.strMeal}</p>
+        `;
+        btn.innerText = 'Comments';
+        btn2.innerText = 'Reservations';
+        section.classList = 'food_items flex';
+        div.classList = 'column flex';
+        btn.classList = 'button';
+        btn2.classList = 'button';
+        twoDiv.classList = 'flex actions_name';
+        likeIcon.classList = 'fa-solid fa-heart like';
+        likeCounts.classList = 'like_count';
+        likeCounts.innerText = '0';
+        threeDiv.append(likeIcon, likeCounts);
+        twoDiv.append(threeDiv);
+        div.append(btn, btn2);
+        section.append(twoDiv, div);
+        main.append(section);
+        const kisplay = async (gameData) => {
+          likeCounts.innerHTML = '';
+          const displayScores = gameData.map((list) => `<div class="new_list">
+                                                            <p> ${list.likes} </p>
+                                                            
+                                                          </div>`).join('');
+          likeCounts.innerHTML = displayScores;
+        };
+
+        const lik = async () => {
+          const like = new Likes(item.idMeal);
+          const response = await fetch(likeUrl, { method: 'POST', headers: { 'content-type': 'application/json' }, body: JSON.stringify(like) });
+          const data = response;
+          return data;
+        };
+
+        const get = async (id) => {
+          const response = await fetch(`${likeUrl}?item_id=${id}`);
+          const data = await response.json(id);
+          const idss = data.filter((ids) => ids.item_id === id);
+          if (response.ok) {
+            kisplay(idss);
+          }
+        };
+        get(item.idMeal);
+        threeDiv.addEventListener('click', () => {
+          lik();
+          get(item.idMeal);
+        });
```

Comment:
- [ ] We have one small bug that I believe you can solve. Let's take a look at the project requirement below:

  > When the user clicks on the Like button of an item, the interaction is recorded in the Involvement API and the screen is updated.

Right now when one clicks the like button, the screen is not updated. Instead, one has to refresh the page to notice that the like was added. Please let's refactor the code and see that this behavior is satisfied.

Reply:
https://www.loom.com/share/7d1f939852a2450c990e51a8490da1ab

Reply:
@jaamanu, sorry about this. I have realized that is some wait time/lag in the API post and get requests. This is all good now.